File: lib/shared/models/day_status.dart
enum DayStatus {
  green,
  yellow,
  red,
  empty;

  String get label => switch (this) {
        DayStatus.green => 'Выполнено',
        DayStatus.yellow => 'Қаза закрыта',
        DayStatus.red => 'Пропуск',
        DayStatus.empty => 'Нет данных',
      };
}

enum WeekStatus {
  green,
  yellow,
  red,
  empty,
  future,
  current;

  String get label => switch (this) {
        WeekStatus.green => 'Все намазы выполнены',
        WeekStatus.yellow => 'Были пропуски, қаза закрыта',
        WeekStatus.red => 'Есть незакрытая қаза',
        WeekStatus.empty => 'Нет данных',
        WeekStatus.future => 'Будущая неделя',
        WeekStatus.current => 'Текущая неделя',
      };
}

File: QadamIOS/Qadam/Core/DayStatus.swift
import Foundation

enum DayStatus: String, Codable {
    case green
    case yellow
    case red
    case empty

    var label: String {
        switch self {
        case .green: return "Выполнено"
        case .yellow: return "Қаза закрыта"
        case .red: return "Пропуск"
        case .empty: return "Нет данных"
        }
    }
}

enum WeekStatus: String, Codable {
    case green
    case yellow
    case red
    case empty
    case future
    case current

    var label: String {
        switch self {
        case .green: return "Все намазы выполнены"
        case .yellow: return "Были пропуски, қаза закрыта"
        case .red: return "Есть незакрытая қаза"
        case .empty: return "Нет данных"
        case .future: return "Будущая неделя"
        case .current: return "Текущая неделя"
        }
    }
}
